src/gui/foldercreationdialog.cpp

```diff
@@ -0,0 +1,80 @@
+#include "foldercreationdialog.h"
+#include "ui_foldercreationdialog.h"
+
+#include <limits>
+
+#include <QDir>
+#include <QMessageBox>
+
+namespace  {
+  const QString SuggestedFolderNamePrefix = QObject::tr("New folder");
+};
+
+FolderCreationDialog::FolderCreationDialog(QWidget *parent) :
+    QDialog(parent),
+    ui(new Ui::FolderCreationDialog)
+{
+    ui->setupUi(this);
+
+    ui->labelErrorMessage->setVisible(false);
+
+    setWindowFlags(windowFlags() & ~Qt::WindowContextHelpButtonHint);
+}
+
+FolderCreationDialog::~FolderCreationDialog()
+{
+    delete ui;
+}
+
+void FolderCreationDialog::setDestination(const QString& destination)
+{
+    Q_ASSERT(!destination.endsWith('/'));
+    _destination = destination;
+}
+
+void FolderCreationDialog::showEvent(QShowEvent *event)
+{
+    const auto newFolderFullPath = _destination + "/" + SuggestedFolderNamePrefix;
+    if (!QDir(newFolderFullPath).exists()) {
+        ui->newFolderNameEdit->setText(SuggestedFolderNamePrefix);
+    } else {
+        for (unsigned int i = 2; i < std::numeric_limits<unsigned int>::max(); ++i) {
+            const QString suggestedPostfix = QString(" (%1)").arg(i);
+
+            if (!QDir(newFolderFullPath + suggestedPostfix).exists()) {
+                ui->newFolderNameEdit->setText(SuggestedFolderNamePrefix + suggestedPostfix);
+                break;
+            }
+        }
+    }
+
+    ui->newFolderNameEdit->setFocus();
+    ui->newFolderNameEdit->selectAll();
+
+    QDialog::showEvent(event);
+}
+
+void FolderCreationDialog::accept()
+{
+    if (QDir(_destination + "/" + ui->newFolderNameEdit->text()).exists()) {
+        ui->labelErrorMessage->setVisible(true);
+        return;
+    }
+
+    if (!QDir(_destination).mkdir(ui->newFolderNameEdit->text())) {
+        QMessageBox messageBox;
+        messageBox.critical(0, tr("Error"), tr("Could not create a folder! Check your write permissions."));
+        messageBox.setFixedSize(500,200);
+    }
+
+    QDialog::accept();
+}
+
+void FolderCreationDialog::on_newFolderNameEdit_textEdited(const QString &)
```

Comment:
Maybe a matter of personal taste... but I think also a question of benefiting from compile time checks. I'd not rely on such connect via convention but have my slot named on my own convention and use and explicit connect in the ctor.

This way if it should break because something changed in the ui file (for instance) the compiler would catch it.